src/Raven.Client/Util/StreamWithTimeout.cs

```diff
@@ -137,23 +146,16 @@ public override int Read(byte[] buffer, int offset, int count)
             return AsyncHelpers.RunSync(() => ReadAsyncWithTimeout(buffer, offset, count, CancellationToken.None));
         }
 
-        public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            if (_canBaseStreamTimeoutOnRead)
-            {
-                var read = await _stream.ReadAsync(buffer, offset, count, cancellationToken).ConfigureAwait(false);
-                _totalRead += read;
-                return read;
-            }
-
             // _totalRead is counted in ReadAsyncWithTimeout
-            return await ReadAsyncWithTimeout(buffer, offset, count, cancellationToken).ConfigureAwait(false);
+            return ReadAsyncWithTimeout(buffer, offset, count, cancellationToken);
         }
 
         private async Task<int> ReadAsyncWithTimeout(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
             _readCts?.Dispose();
-            _readCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            _readCts = _readCts == default ? new CancellationTokenSource() : CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```

Comment:
what happens if `cancellationToken` is passed to this method and `_readCts`  is null ? shouln't we create a `nkedTokenSource` in such case?

Reply:
Yes. My bad. Is supposed to be `cancellationToken == default`